src/odemis/util/comp.py

```diff
@@ -162,22 +162,16 @@ def generate_zlevels(focuser: "Actuator", zrange: Tuple[float, float], zstep: fl
     if n > MAX_ZLEVELS:
         raise IndexError(f"The number of zlevels, {n}, is too large. Reduce the zstep value to < {MAX_ZLEVELS}.")
 
-    # zstep often doesn't exactly fit to have a round number of zlevels. So try
-    # the floor and ceil values for the number of zlevels, and take the one that
-    # give a step closest to the requested zstep.
+    # When the zstep is bigger than the z range
+    # we return the z range itself
     n_f = math.floor(n)
-    n_c = math.ceil(n)
-
-    if n_f <= 1:  # would cause a division by zero
-        # => do the minimum amount of steps
-        n = 2
-    else:
-        # errors
-        ef = abs((zrange_abs[1] - zrange_abs[0]) / (n_f - 1) - abs(zstep))
-        ec = abs((zrange_abs[1] - zrange_abs[0]) / (n_c - 1) - abs(zstep))
-        n = n_f if ef < ec else n_c
-
-    if zstep > 0:
-        return numpy.linspace(zrange_abs[0], zrange_abs[1], n).tolist()
-    elif zstep < 0:
-        return numpy.linspace(zrange_abs[1], zrange_abs[0], n).tolist()
+    if n_f <= 1:
+        return [zrange_abs[0], zrange_abs[1]]
```

Comment:
In this case it's not based on the current focus position, right?

Reply:
I don't think there is a lot better to do. Firstly, that's an extremely unlikely corner case (either the focuser can barely move, which is very odd, or the user asks for gigantic zstep, which is odd).
The alternatives would be to return just the current pos (quite sad for a zstack), or to return the extrema + current pos (3 points not at equal distance from each other, which is odd). Thera, would you suggest something else?

Reply:
No, just that now the docstring mentions "The zlevels are based on the current focus position.", I would suggest to update that to mention that in unlikely cases for very large z-step or very small rang it returns the min/max range as z-levels.